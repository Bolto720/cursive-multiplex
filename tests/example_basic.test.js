const { setup } = require('shellshot');

setup();

it(
    'checks if the basic example produces the same output',
    async () => {
        await expect.command('timeout 1 script --quiet --command ../target/debug/examples/basic /dev/null')
            .withEnv({
                TERM: "xterm-256color",
                LINES: 24,
                COLUMNS: 80,
            })
            .forStdout(expectation => expectation.toMatchSnapshot());
    },
);
