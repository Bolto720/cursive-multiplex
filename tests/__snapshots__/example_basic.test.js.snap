// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if the basic example produces the same output 1`] = `
"[?1049h[22;0;0t[1;24r(B[m[4l[?7h[?1h=[?1006;1000h[39;49m[?25l[?1002h[39;49m[34m[44m[H[2J[1;39H[30m[47m[1K [37m[44m [30m[47m (B[0;7m[34m[47m [37b [2;1H(B[0m[30m[47mLorem ipsum dolor sit amet, [11b[37m[44m [30m[47m (B[0;7m[34m[47m [37b [3;1H(B[0m[30m[47mconsectetur adipiscing elit. Donec     [37m[44m [30m[47m (B[0;7m[34m[47m [37b [4;1H(B[0m[30m[47mcongue porttitor pellentesque. [8b[37m[44m [30m[47m (B[0;7m[34m[47m [37b [5;1H(B[0m[30m[47mVestibulum a tellus sagittis, blandit  [37m[44m [30m[47m (B[0;7m[34m[47m [37b [6;1H(B[0m[30m[47merat ac, finibus eros. Praesent cursus [37m[44m [30m[47m (B[0;7m[34m[47m [37b [7;1H(B[0m[30m[47mat ligula laoreet congue. Proin [7b| (B[0;7m[34m[47m [37b [8;1H(B[0m[30m[47mvehicula diam mattis metus aliquet     | (B[0;7m[34m[47m [37b [9;1H(B[0m[30m[47maliquam. Nullam finibus tellus id      | (B[0;7m[34m[47m [37b [10;1H(B[0m[30m[47mdolor porta venenatis. Cras vestibulum | (B[0;7m[34m[47m [37b [11;1H(B[0m[30m[47mleo sit amet congue ultrices. [9b| (B[0;7m[34m[47m [37b [12;1H(B[0m[30m[47mPhasellus convallis ut enim tincidunt  |  ~T~@ ~T~@ ~T~@ ~T~@ ~T~@ ~T~@ ~T~@ ~T~[13;1H@ ~T~@ ~T~@ ~T~@ ~T~@ ~T~@ ~T~@ ~T~@ ~T~@[34m[47m_(B[0;7m[34m[47m [36b [14;1H(B[0m[30m[47m@ ~T~@ ~T~@ ~T~@ ~T~@ ~T~@ ~T~@ ~T~@ ~T~@(B[0;7m[34m[47m [37b [15;1H(B[0m[30m[47mIn velit felis, consectetur quis [6b| (B[0;7m[34m[47m [37b [16;1H(B[0m[30m[47mfringilla id, interdum congue metus.   | (B[0;7m[34m[47m [37b [17;1H(B[0m[30m[47mMauris tincidunt, nibh a fermentum     | (B[0;7m[34m[47m [37b [18;1H(B[0m[30m[47mposuere, nibh elit auctor lacus, [6b| (B[0;7m[34m[47m [37b [19;1H(B[0m[30m[47msollicitudin lobortis nisi arcu quis   | (B[0;7m[34m[47m [37b [20;1H(B[0m[30m[47mmassa. Ut id augue malesuada justo     | (B[0;7m[34m[47m [37b [21;1H(B[0m[30m[47mvenenatis pellentesque. Donec egestas  | (B[0;7m[34m[47m [37b [22;1H(B[0m[30m[47mnec purus sit amet euismod. Integer    | (B[0;7m[34m[47m [37b [23;1H(B[0m[30m[47maliquet sollicitudin ex id viverra.    | (B[0;7m[34m[47m [37b [24;1H(B[0m[30m[47m  Hello from cursive_multiplex  Feel free to try out the examples simply with \`[?7lc[?7h[39;49m(B[m[?1006;1000l[24;1H[?12l[?25h[?1049l[23;0;0t
[?1l>"
`;
